CMakeLists.txt

```diff
@@ -26,13 +26,56 @@ if(POLICY CMP0071)
 endif()
 
 # Set up vcpkg
-if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED VCPKG_ROOT)
-    set(VCPKG_ROOT "$ENV{VCPKG_ROOT}")
+if(WIN32 OR APPLE)
+  option(VCPKG "Use vcpkg for dependencies" ON)
+else()
+  option(VCPKG "Use vcpkg for dependencies" OFF)
 endif()
 
-if(DEFINED VCPKG_ROOT)
+if(VCPKG)
+  message(STATUS "Using dependencies from vcpkg")
+
   set(X_VCPKG_APPLOCAL_DEPS_INSTALL TRUE)
 
+  if(NOT DEFINED ENV{VCPKG_BINARY_SOURCES})
+    set(ENV{VCPKG_BINARY_SOURCES} "clear;nuget,mixxx-github-packages,read;default;")
+  endif()
+
+  set(ENV{VCPKG_DISABLE_COMPILER_TRACKING} "true")
```

Comment:
This must be only active for local builds. We want to rebuild the packages on the CI if the compiler version changes. 
Imagine a compile bug is fixed. We don't want to have that in our release builds. For local builds the user can decide to trust the CI or rebuild vcpkg with there own compiler version.

Reply:
> For local builds the user can decide to trust the CI or rebuild vcpkg with there own compiler version.

This is not correct. If VCPKG_DISABLE_COMPILER_TRACKING is not used on CI, local builds will never have cache hits from CI packages.

Reply:
Ah OK, makes sense.